tests/System.IO.Pipelines.Tests/UnownedBufferReaderFacts.cs

```diff
@@ -333,9 +326,21 @@ public class UnownedBufferReaderFacts
                 Assert.Equal("Hello ", Encoding.UTF8.GetString(preserved.Buffer.ToArray()));
             }
 
+            EnsureSpanDisposed(preserved.Buffer.First);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
```

Comment:
It could in release builds, if the compiler can determine that accessing First and Span are side-effect free.